src/scheduler/limits.c

```diff
@@ -2022,22 +2024,35 @@ check_queue_max_run_soft(server_info *si, queue_info *qi, resource_resv *rr)
 {
 	int	max_running;
 	char	*key;
+	counts	*cnt = NULL;
+	int used = 0;
+
 
 	if (qi == NULL)
 		return (PREEMPT_TO_BIT(PREEMPT_ERR));
 
+	if (!qi->has_all_limit)
+	    return (0);
+
 	if ((key = entlim_mk_runkey(LIM_OVERALL, allparam)) == NULL)
 		return (PREEMPT_TO_BIT(PREEMPT_ERR));
 	max_running = (int) lim_get(key, LI2RUNCTXSOFT(qi->liminfo));
 	free(key);
 
 	if ((max_running == SCHD_INFINITY) ||
 		(max_running > qi->sc.running)) {
-		qi->soft_limit_preempt_bit = 0;
+		qi->alljobcounts->soft_limit_preempt_bit = 0;
 		return (0);
 	} else {
-		qi->soft_limit_preempt_bit = PREEMPT_TO_BIT(PREEMPT_OVER_QUEUE_LIMIT);
-		return (PREEMPT_TO_BIT(PREEMPT_OVER_QUEUE_LIMIT));
+		/* at this point, we know a limit is set  for PBS_ALL*/
```

Comment:
Is the above code necessary?  We will return if the queue doesn't have a soft limit on line 2034. (and in other places).  Aso, there is two spaces between set and for, and no space between PBS_ALL and the end comment. (here and other places)